mr-plonky2-circuits/src/query_erc20/storage/leaf.rs

```diff
@@ -1,60 +1,84 @@
 
 #[derive(Serialize, Deserialize)]
 pub struct LeafWires {
-    pub packed_mapping_key: Array<U32Target, PACKED_MAPPING_KEY_LEN>,
-    pub packed_mapping_value: Array<U32Target, PACKED_VALUE_LEN>,
+    address: PackedAddressTarget,
+    value: PackedU256Target,
+    total_supply: PackedU256Target,
+    reward: PackedU256Target,
 }
 
-/// This circuit prove the new root hash of a leaf containing the requested data
 #[derive(Clone, Debug)]
 pub struct LeafCircuit {
-    pub mapping_key: [u32; PACKED_MAPPING_KEY_LEN],
-    pub mapping_value: [u32; PACKED_VALUE_LEN],
+    pub address: Address,
+    pub value: U256,
+    pub total_supply: U256,
+    pub reward: U256,
 }
 
 impl LeafCircuit {
     pub fn assign(&self, pw: &mut PartialWitness<GoldilocksField>, wires: &LeafWires) {
-        wires
-            .packed_mapping_key
-            .assign_from_data(pw, &self.mapping_key);
-        wires
-            .packed_mapping_value
-            .assign_from_data(pw, &self.mapping_value);
+        let address = (&self.address.0).pack().try_into().unwrap();
+        wires.address.assign_from_data(pw, &address);
+
+        let mut bytes = [0; 32];
+        [
+            (self.value, &wires.value),
+            (self.total_supply, &wires.total_supply),
+            (self.reward, &wires.reward),
+        ]
+        .iter()
+        .for_each(|(v, w)| {
+            v.to_little_endian(&mut bytes);
+            let v = bytes.pack().try_into().unwrap();
+
+            w.assign_from_data(pw, &v);
+        });
     }
 
     fn build(b: &mut CircuitBuilder<GoldilocksField, 2>) -> LeafWires {
-        let key_u32 = PackedMappingKeyTarget::new(b);
-        let value_u32 = Array::<U32Target, PACKED_VALUE_LEN>::new(b);
-        let kv = key_u32.concat(&value_u32).to_targets();
-
-        // the digest is done on the key only, in compact form, because our goal is
-        // to reval all the keys at the last step of the computation graph
-        let digest = b.map_to_curve_point(&key_u32.to_targets().arr);
-        // the root is done on both as this is what proves the inclusion in the storage db
-        let root = b.hash_n_to_hash_no_pad::<PoseidonHash>(Vec::from(kv.arr));
+        let address = PackedAddressTarget::new(b);
+        let [value, total_supply, reward] = [0; 3].map(|_| PackedU256Target::new(b));
+
+        // C = poseidon("LEAF" || pack_u32(address) || pack_u32(value))
+        let prefix: Vec<_> = HASH_PREFIX
+            .iter()
+            .map(|v| GoldilocksField::from_canonical_u8(*v))
+            .collect();
+        let prefix = b.constants(&prefix);
+        let inputs = prefix
+            .into_iter()
+            .chain(address.arr.map(|v| v.to_target()))
+            .chain(value.arr.map(|v| v.to_target()))
+            .collect();
+        let c = b.hash_n_to_hash_no_pad::<PoseidonHash>(inputs);
+
+        // V = R * value / totalSupply
+        // TODO: U256 operations
+        let v = &reward;
```

Comment:
v should be v. rewards should be rewards.

Reply:
Fixed to `PackedU256Target::new(b)` in commit [bc67b19](https://github.com/Lagrange-Labs/Euclid-database/pull/6/commits/bc67b192a05d040164a52a8ce9ba9bdcf952a573).